File: Projeto_5/src/SensorGPS.cpp
#include "SensorGPS.h"
#include <iostream>

SensorGPS::SensorGPS(std::string id, int ts, double lat, double lon) 
    : Dispositivo(id, ts), latitude(lat), longitude(lon) {}

void SensorGPS::processarDados() {
    std::cout << "[GPS] ID: " << idDispositivo << " | Tempo: " << timestamp 
              << " | Coordenadas: (" << latitude << ", " << longitude << ")" << std::endl;
}

void SensorGPS::transmitirPayload() {
    std::cout << " -> Transmitindo payload GPS (Texto Puro)..." << std::endl;
}

void SensorGPS::transmitirPayload(std::string chaveCripto) {
    std::cout << " -> Transmitindo payload GPS (Criptografado com chave: " << chaveCripto << ")" << std::endl;
}

File: Projeto_5/models/SensorGPS.py
from models.Dispositivo import Dispositivo

class SensorGPS(Dispositivo):
    def __init__(self, id_dispositivo: str, timestamp: int, latitude: float, longitude: float):
        Dispositivo.__init__(self, id_dispositivo, timestamp)
        self.__latitude = latitude    
        self.__longitude = longitude  

    def processarDados(self) -> None:
        print(f"[ RODANDO... ] GPS Processando {self._idDispositivo} | Lat: {self.__latitude}, Lon: {self.__longitude}")

    def transmitirPayload(self, chave_cripto: str = None) -> None:
        if chave_cripto is None:
            print(" -> Enviando payload em texto puro (Inseguro).")
        else:
            print(f" -> Enviando payload criptografado com chave [{chave_cripto}].")